source/extensions/lightspeed.layer_manager.core/lightspeed/layer_manager/core/core.py

```
@@ -91,6 +91,9 @@ def get_loaded_project_with_data_models(self) -> LayerResponseModel:
     def open_project_with_data_models(self, params: OpenProjectPathParamModel):
         self.__context.open_stage(str(params.layer_id))
 
+    def close_project_with_data_models(self):
```

We would want a force flag here, so that we can ignore any pending changes in the scene. Otherwise, it would be good to throw a 403 (Forbidden) error when closing out with pending changes.

---

added save force before close

---

Example of an exception raising method. You don't have to use the wrapper, you could just raise the exception directly :P

https://github.com/NVIDIAGameWorks/toolkit-remix/blob/main/source/extensions/omni.flux.service.factory/omni/flux/service/factory/services/base.py#L159

Basically we just don't want an external tool to accidentally discard all pending changes in a stage, so returning an error when there are pending changes unless the user forced the request allows us to make that explicit